### PR TITLE
refactor: [Rector] Apply ExplicitBoolCompareRector

### PR DESCRIPTION
**Pull request by samsonasik**
**Description**

Make if conditions more explicit and reduce phpstan baseline errors.

**Checklist:**
- [x] Securely signed commits
- [ ] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Database/OCI8/Builder.php [hunk 1]
```diff
@@ -222,7 +222,7 @@ protected function _limit(string $sql, bool $offsetIgnore = false): string
         }
 
         $this->limitUsed    = true;
-        $limitTemplateQuery = 'SELECT * FROM (SELECT INNER_QUERY.*, ROWNUM RNUM FROM (%s) INNER_QUERY WHERE ROWNUM < %d)' . ($offset ? ' WHERE RNUM >= %d' : '');
+        $limitTemplateQuery = 'SELECT * FROM (SELECT INNER_QUERY.*, ROWNUM RNUM FROM (%s) INNER_QUERY WHERE ROWNUM < %d)' . ($offset !== 0 ? ' WHERE RNUM >= %d' : '');
```
Comment (samsonasik) [inline system/Database/OCI8/Builder.php:225]: negative is truthy so `!==0` is used, ref https://3v4l.org/dajZg

**Comment by samsonasik** [comment]
Ready to review 👍

**Review by paulbalandan — COMMENTED** [review]
I'm not sure abt the change in strings to `=== ''` and `=== '0'`.  Are we really checking for non-falsy-string? Or maybe just non-empty-string?

For the others, I think I'm good with those changes.

**Comment by samsonasik** [comment]
This is the exact equal truthy check to string, which should be safer.

 if it only check on empty string `""`, I can change

File: system/CLI/CLI.php [hunk 2]
```diff
@@ -262,7 +262,7 @@ public static function prompt(string $field, $options = null, $validation = null
             $default = $options[0];
         }
 
-        static::fwrite(STDOUT, $field . (trim($field) ? ' ' : '') . $extraOutput . ': ');
+        static::fwrite(STDOUT, $field . (trim($field) !== '' && trim($field) !== '0' ? ' ' : '') . $extraOutput . ': ');
```
Comment (kenjis) [inline system/CLI/CLI.php:265]: Remove ` && trim($field) !== '0'`. It is kind of bug.

The current behavior.

`What is your favorite color?`:
```console
$ ./spark test

CodeIgniter v4.4.7 Command Line Tool - Server Time: 2024-04-03 22:54:23 UTC+00:00

What is your favorite color? : a
```

`''`:
```console
$ ./spark test

CodeIgniter v4.4.7 Command Line Tool - Server Time: 2024-04-03 22:54:39 UTC+00:00

: a
```

`' '`:
```console
$ ./spark test

CodeIgniter v4.4.7 Command Line Tool - Server Time: 2024-04-03 22:54:48 UTC+00:00

 : a
```

`'0'`:
```console
$ ./spark test

CodeIgniter v4.4.7 Command Line Tool - Server Time: 2024-04-03 22:56:09 UTC+00:00

0: a
```

File: system/Config/DotEnv.php [hunk 3]
```diff
@@ -140,7 +140,7 @@ public function normaliseVariable(string $name, string $value = ''): array
      */
     protected function sanitizeValue(string $value): string
     {
-        if (! $value) {
+        if ($value === '' || $value === '0') {
```
Comment (kenjis) [inline system/Config/DotEnv.php:143]: This is not a bug, but no need to treat only `'0'`.

File: system/Database/OCI8/Builder.php (hunk 1 above)
Comment (kenjis) [reply system/Database/OCI8/Builder.php:225]: `$offset` is non negative.

File: system/HTTP/DownloadResponse.php [hunk 4]
```diff
@@ -194,7 +194,7 @@ private function getContentDisposition(): string
 
         $result = sprintf('attachment; filename="%s"', $downloadFilename);
 
-        if ($utf8Filename) {
+        if ($utf8Filename !== '' && $utf8Filename !== '0') {
```
Comment (kenjis) [inline system/HTTP/DownloadResponse.php:197]: Remove `&& $utf8Filename !== '0'`.
Filename `0` is valid.

File: system/Helpers/form_helper.php [hunk 5]
```diff
@@ -29,7 +29,7 @@
     function form_open(string $action = '', $attributes = [], array $hidden = []): string
     {
         // If no action is provided then set to the current url
-        if (! $action) {
+        if ($action === '' || $action === '0') {
```
Comment (kenjis) [inline system/Helpers/form_helper.php:32]: Remove the case of `0`. It is a valid URI path.

File: system/Config/DotEnv.php (hunk 3 above)
Comment (samsonasik) [reply system/Config/DotEnv.php:143]: do you mean only check empty string `''` ?

File: system/Helpers/url_helper.php [hunk 6]
```diff
@@ -318,7 +318,7 @@ function safe_mailto(string $email, string $title = '', $attributes = ''): strin
 
         // improve obfuscation by eliminating newlines & whitespace
         $cspNonce = csp_script_nonce();
-        $cspNonce = $cspNonce ? ' ' . $cspNonce : $cspNonce;
+        $cspNonce = $cspNonce !== '' && $cspNonce !== '0' ? ' ' . $cspNonce : $cspNonce;
```
Comment (kenjis) [inline system/Helpers/url_helper.php:321]: Remove `0`. The same as above.

File: system/Router/RouteCollection.php [hunk 7]
```diff
@@ -767,7 +767,7 @@ public function group(string $name, ...$params)
         // To register a route, we'll set a flag so that our router
         // will see the group name.
         // If the group name is empty, we go on using the previously built group name.
-        $this->group = $name ? trim($oldGroup . '/' . $name, '/') : $oldGroup;
+        $this->group = $name !== '' && $name !== '0' ? trim($oldGroup . '/' . $name, '/') : $oldGroup;
```
Comment (kenjis) [inline system/Router/RouteCollection.php:770]: Remove `0`. The group name (= URI path) `0` is valid.

File: system/Session/Handlers/FileHandler.php [hunk 8]
```diff
@@ -73,7 +73,7 @@ public function __construct(SessionConfig $config, string $ipAddress)
         } else {
             $sessionPath = rtrim(ini_get('session.save_path'), '/\\');
 
-            if (! $sessionPath) {
+            if ($sessionPath === '' || $sessionPath === '0') {
```
Comment (kenjis) [inline system/Session/Handlers/FileHandler.php:76]: Remove `0`. 
`0` seems invalid as a file path, but no need to treat specially.

File: system/Config/DotEnv.php (hunk 3 above)
Comment (kenjis) [reply system/Config/DotEnv.php:143]: Yes, remove `|| $value === '0'`.

File: system/Database/OCI8/Builder.php (hunk 1 above)
Comment (samsonasik) [reply system/Database/OCI8/Builder.php:225]: I check on another part on `BaseBuilder`, `$offset !== 0` is used so I think `!==0` is valid.

**Comment by samsonasik** [comment]
@kenjis I updated to remove `!== "0"` check 👍

File: system/Database/OCI8/Builder.php (hunk 1 above)
Comment (kenjis) [reply system/Database/OCI8/Builder.php:225]: `$offset` is non negative. So `!== 0` is no problem.
But it is the same as `> 0`.

**Review by kenjis — APPROVED** [review]

File: system/Database/OCI8/Builder.php (hunk 1 above)
Comment (samsonasik) [reply system/Database/OCI8/Builder.php:225]: let use `!==0` for consistency

**Comment by samsonasik** [comment]
Ready to merge 👍

**Review by paulbalandan — APPROVED** [review]

**Comment by samsonasik** [comment]
Let's merge it, thank you @paulbalandan @kenjis for the review 👍

**Pull request merged**